### PR TITLE
[daint-gpu] gromacs with cdt 20.08 and pat

### PR DESCRIPTION
**Pull request by sebkelle1**

### CODE REVIEW

**Comment by sebkelle1** [comment]
restest this please

**Comment by sebkelle1** [comment]
retest this please

File: easybuild/easyconfigs/g/GROMACS/GROMACS-2020.3-CrayGNU-20.08-cuda-pat7.1.1.eb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+# $Id$
+#
+# This work implements a part of the HPCBIOS project and is a component of the policy:
+# http://hpcbios.readthedocs.org/en/latest/HPCBIOS_2012-93.html
+#
+easyblock = 'CMakeMake'
+name = 'GROMACS'
+version = "2020.3"
+local_patversion = '7.1.1'
+versionsuffix = '-cuda-pat%s' % (local_patversion)
```
Comment (lucamar) [inline easybuild/easyconfigs/g/GROMACS/GROMACS-2020.3-CrayGNU-20.08-cuda-pat7.1.1.eb:18]:
```suggestion
versionsuffix = '-cuda-pat'
```

File: easybuild/easyconfigs/g/GROMACS/GROMACS-2020.3-CrayGNU-20.08-cuda-pat7.1.1.eb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+toolchain = {'name': 'CrayGNU', 'version': '20.08'}
+toolchainopts = {'usempi': True, 'openmp': True, 'pic': True, 'verbose': False, 'opt': True, 'dynamic': False}
+
+source_urls = ['ftp://ftp.gromacs.org/pub/gromacs/']
+sources = [SOURCELOWER_TAR_GZ]
+
+# CMake dependency with dummy toolchain
+builddependencies = [
+    ('CMake', '3.14.1', '', True),
+    ('perftools-lite/%s' % local_patversion, EXTERNAL_MODULE),
```
Comment (lucamar) [inline easybuild/easyconfigs/g/GROMACS/GROMACS-2020.3-CrayGNU-20.08-cuda-pat7.1.1.eb:33]:
```suggestion
    ('CMake', '3.14.5', '', True),
    ('perftools-lite', EXTERNAL_MODULE),
```

File: easybuild/easyconfigs/g/GROMACS/GROMACS-2020.3-CrayGNU-20.08-cuda-pat7.1.1.eb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+]
+
+dependencies = [
+    ('craype-accel-nvidia60', EXTERNAL_MODULE),
+    ('cray-libsci', EXTERNAL_MODULE)
```
Comment (lucamar) [inline easybuild/easyconfigs/g/GROMACS/GROMACS-2020.3-CrayGNU-20.08-cuda-pat7.1.1.eb:38]:
```suggestion
    ('craype-accel-nvidia60', EXTERNAL_MODULE),
```

**Review by lucamar — CHANGES_REQUESTED** [review]
Thanks @sebkelle1, I have added some suggested edits, based on the instructions that I have provided in https://jira.cscs.ch/browse/UES-1134. 
I have also changed `CMake` to the default, which should work as well. Do you really need `craype-accel-nvidia60` as runtime dependency? In general, `cudatoolkit` in `builddependencies` should be enough.

File: easybuild/easyconfigs/g/GROMACS/GROMACS-2020.3-CrayGNU-20.08-cuda-pat7.1.1.eb (hunk 3 above)
Comment (lucamar) [reply easybuild/easyconfigs/g/GROMACS/GROMACS-2020.3-CrayGNU-20.08-cuda-pat7.1.1.eb:38]: The module `cray-libsci` is loaded by default with `PrgEnv-gnu`, loaded by `CrayGNU`.

**Comment by lucamar** [comment]
The build failed with the following error message:
```
ERROR: Program '/tmp/jenscscs/ad65d21gpu/GROMACS/2020.3/CrayGNU-20.08-cuda-pat7.1.1/easybuild_obj/share/template/../../bin/template+orig' is already instrumented with PerfToolsWARNING: PerfTools is saving object files from a temporary directory into directory '/users/jenscscs/.craypat/template/12138'
WARNING: the link is executed in a temporary directory which may not be accessible later for pat_build
WARNING: the resulting executable was created in a temporary directory which may not be accessible later for pat_build
INFO: creating the PerfTools-instrumented executable '../../bin/template' (lite-samples) ...FAILED
collect2: error: ld returned 1 exit status
```
See the log file `/scratch/snx3000/jenscscs/TestingEB/daint-gpu/ad65d21-4220/tmp/eb-n2t4aq/easybuild-GROMACS-2020.3-20200918.123940.DHkWU.log`, referenced in https://jenkins.cscs.ch/blue/organizations/jenkins/TestingEB/detail/TestingEB/4220/pipeline.

**Comment by sebkelle1** [comment]
addressed all requests that could be satisfied

**Review by victorusu — APPROVED** [review]
`lgtm`

**Review by lucamar — APPROVED** [review]
Thanks @sebkelle1 and @victorusu

**Pull request merged**